### PR TITLE
Merge properties files into neo4j.properties

### PR DESCRIPTION
**Pull request by benbc**

### CODE REVIEW

**Comment by benbc** [comment]
@jakewins Any chance of a review of this, please, since @srbaker is away this week? Please don't merge as I need to coordinate the merge with the testing tools and the milestone release.

File: community/server/pom.xml [hunk 1]
```diff
@@ -388,8 +388,8 @@
           <classpathScope>test</classpathScope>
           <systemProperties>
             <systemProperty>
-              <key>org.neo4j.server.properties</key>
-              <value>${basedir}/neo4j-home/conf/neo4j-server.properties</value>
+              <key>org.neo4j.config.file</key>
+              <value>${basedir}/neo4j-home/conf/neo4j.properties</value>
```
Comment (jakewins) [inline community/server/pom.xml:392]: Entirely outside this PR: Probably worth reviewing if we still need that neo4j-home folder and the associated config. Ideally all the server tests should be confined to things that go under `target/` nowadays. The neo4j-home checked into the source tree is a very, very old annoyance.

File: community/server/src/docs/ops/server-configuration.asciidoc [hunk 2]
```diff
@@ -3,15 +3,14 @@
 
 .Quick info
 ***********
-* The server's primary configuration file is found under _conf/neo4j-server.properties_.
-* Low-level performance tuning parameters and configuration of legacy indexes and the remote shell are found in _conf/neo4j.properties_.
+* Neo4j's configuration is found in _conf/neo4j.properties_.
```
Comment (jakewins) [inline community/server/src/docs/ops/server-configuration.asciidoc:6]: It seems like the advice here would not apply to most actual packaged variants of Neo4j, or? Perhaps this should say something like "the location of the Neo4j config file depends on how you installed Neo4j, common locations are blah blah and you can control which location is used via blah blah mechanism"

File: community/server/src/test/java/org/neo4j/server/helpers/CommunityServerBuilder.java [hunk 3]
```diff
@@ -76,26 +70,13 @@
...
-            return new ImpermanentGraphDatabase( storeDir, params, GraphDatabaseDependencies.newDependencies(dependencies) );
-        }
+    public static LifecycleManagingDatabase.GraphFactory IN_MEMORY_DB = ( config, dependencies ) -> {
+        File storeDir = config.get( ServerSettings.legacy_db_location );
+        Map<String, String> params = config.getParams();
+        params.put( CommunityFacadeFactory.Configuration.ephemeral.name(), "true" );
+        return new ImpermanentGraphDatabase( storeDir, params, GraphDatabaseDependencies.newDependencies(dependencies) );
     };
 
-    private enum WhatToDo
```
Comment (jakewins) [inline community/server/src/test/java/org/neo4j/server/helpers/CommunityServerBuilder.java:91]: Thank you :)

File: packaging/neo4j-desktop/src/main/java/org/neo4j/desktop/config/unix/UnixInstallation.java [hunk 4]
```diff
@@ -48,9 +48,9 @@ public File getVmOptionsFile()
     }
 
     @Override
-    public File getServerConfigurationsFile()
+    public File getConfigurationsFile()
     {
-        return new File( getConfigurationDirectory(), "." + NEO4J_SERVER_PROPERTIES_FILENAME );
+        return new File( getConfigurationDirectory(), "." + NEO4J_PROPERTIES_FILENAME );
```
Comment (jakewins) [inline packaging/neo4j-desktop/src/main/java/org/neo4j/desktop/config/unix/UnixInstallation.java:53]: I know there's been trickery around config files in Desktop before (and literally, that is all I know about it) - do we know that this `NEO4J_PROPERTIES_FILENAME` file is in an editable location that users can find?

**Comment by jakewins** [comment]
I think this looks awesome overall, and I'm really excited about it :) Beyond the small inline notes, I have one question: We renamed the system property to specify where to find the config file, but I couldn't find a test with the equivalent change. Do we have coverage for that system setting? If not, I'd love that as an addition, since it's an important parameter to ensure we don't break in the future.

File: community/server/pom.xml (hunk 1 above)
Comment (benbc) [inline community/server/pom.xml:392]: @jakewins We'll be doing work later to change the way paths for stuff are specified. That'll be a good time to look at this.

File: community/server/src/docs/ops/server-configuration.asciidoc (hunk 2 above)
Comment (benbc) [inline community/server/src/docs/ops/server-configuration.asciidoc:6]: That's a good point. There's a far more general point there. I'll treat it as orthogonal to this change and bring it up with the Docs team.

File: packaging/neo4j-desktop/src/main/java/org/neo4j/desktop/config/unix/UnixInstallation.java (hunk 4 above)
Comment (benbc) [inline packaging/neo4j-desktop/src/main/java/org/neo4j/desktop/config/unix/UnixInstallation.java:53]: Yes, that file was already there and editable.

**Comment by benbc** [comment]
@jakewins I think you're right that there isn't a test for that. This is another area that is going to change again before 3.0. I'd actually like to get rid of using system properties for stuff like this (or indeed for anything at all) and use command line arguments instead for stuff that can't be specified in config. So we'll look at it then.

**Comment by jakewins** [comment]
Then I'm happy.

**Pull request merged**